superbench/benchmarks/micro_benchmarks/micro_base.py

```diff
@@ -70,21 +71,139 @@ def _process_numeric_result(self, metric, result):
         self._result.add_result(metric, sum(result) / len(result))
         return True
 
-    def _process_raw_result(self, raw_output):
+    def print_env_info(self):
+        """Print environments or dependencies information."""
+        # TODO: will implement it when add real benchmarks in the future.
+        pass
+
+
+class MicroBenchmarkWithInvoke(MicroBenchmark):
+    """The base class of micro-benchmarks that need to invoke subprocesses."""
+    def __init__(self, name, parameters=''):
+        """Constructor.
+
+        Args:
+            name (str): benchmark name.
+            parameters (str): benchmark parameters.
+        """
+        super().__init__(name, parameters)
+
+        # Command lines to launch the micro-benchmarks.
+        self._commands = list()
+
+        # Binary name of the current micro-benchmark.
+        self._bin_name = None
+
+    def add_parser_arguments(self):
+        """Add the specified arguments."""
+        super().add_parser_arguments()
+
+        self._parser.add_argument(
+            '--bin_path',
+            type=str,
+            default=None,
+            required=False,
+            help='Specify the binary path of the benchmark.',
+        )
+
+    def _set_binary_path(self):
+        """Search the binary from self._args.bin_path or from system environment path and set the binary path.
+
+        If self._args.bin_path is specified, the binary is only searched inside it. Otherwise, the binary is searched
+        from system environment path.
+
+        Return:
+            True if the binary exists.
+        """
+        if self._bin_name is None:
+            self._result.set_return_code(ReturnCode.MICROBENCHMARK_BINARY_NAME_NOT_SET)
+            logger.error('The binary name is not set - benchmark: {}.'.format(self._name))
+            return False
+
+        existed = True
+        if self._args.bin_path is None:
+            self._args.bin_path = shutil.which(self._bin_name)
+            print(shutil.which(self._bin_name))
+            existed = (self._args.bin_path is not None)
+        else:
+            self._args.bin_path = os.path.join(self._args.bin_path, self._bin_name)
+            existed = os.path.isfile(self._args.bin_path)
+
+        if not existed:
+            self._result.set_return_code(ReturnCode.MICROBENCHMARK_BINARY_NOT_EXIST)
+            logger.error(
+                'The binary does not exist - benchmark: {}, binary name: {}, binary path: {}.'.format(
+                    self._name, self._bin_name, self._args.bin_path
+                )
+            )
+            return False
+
+        return True
+
+    def _preprocess(self):
+        """Preprocess/preparation operations before the benchmarking.
+
+        Return:
+            True if _preprocess() succeed.
+        """
+        if not super()._preprocess():
+            return False
+
+        # Set the environment path.
+        if 'SB_PATH' in os.environ:
```

Comment:
It's only for Microbenchmark or general marco? It would be strange if it's a general usage.

Reply:
It's only for Microbenchmark, yifan suggested the name and the env is set by his module. Do yo have any suggestions?

Reply:
Maybe SB_MICRO_PATH?

Reply:
Is SB_MICRO_PATH OK for you? @abuccts

Reply:
SB_MICRO_PATH